File: models/magic.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Magic = sequelize.define('Magic', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  id_user: {
    type: DataTypes.INTEGER,
    allowNull: false,
    unique: false,
  },
  magias_0: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: true,
  },
  magias_1: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: true,
  },
  magias_2: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: true,
  },
  magias_3: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: true,
  },
  magias_4: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: true,
  },
  magias_5: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: true,
  },
  magias_6: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: true,
  },
  magias_7: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: true,
  },
  magias_8: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: true,
  },
  magias_9: {
    type: DataTypes.ARRAY(DataTypes.INTEGER),
    allowNull: true,
  }
}, {
  timestamps: true,  // Cria automaticamente createdAt e updatedAt
  tableName: 'magic',  // Nome da tabela no banco
});

module.exports = Magic;

File: models/magic.go
package models
import (
	"time"
)
type Magic struct {
	ID      uint   `gorm:"primaryKey;autoIncrement"`
	IDUser  uint   `gorm:"not null"`
	Magias0 []int  `gorm:"type:integer[]"`
	Magias1 []int  `gorm:"type:integer[]"`
	Magias2 []int  `gorm:"type:integer[]"`
	Magias3 []int  `gorm:"type:integer[]"`
	Magias4 []int  `gorm:"type:integer[]"`
	Magias5 []int  `gorm:"type:integer[]"`
	Magias6 []int  `gorm:"type:integer[]"`
	Magias7 []int  `gorm:"type:integer[]"`
	Magias8 []int  `gorm:"type:integer[]"`
	Magias9 []int  `gorm:"type:integer[]"`
	CreatedAt time.Time `gorm:"autoCreateTime;not null"` // Campo auto gerado
	UpdatedAt time.Time `gorm:"autoUpdateTime;not null"`

}